subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ConventionReportTask.java

```diff
@@ -49,9 +51,23 @@
 public abstract class ConventionReportTask extends ConventionTask {
     // todo annotate as required
     private final Transient.Var<Set<Project>> projects = varOf(new HashSet<>(singleton(getProject())));
+    private final DirectoryProperty reportDir;
     private File outputFile;
 
+    /**
+     * Returns the project report directory.
+     *
+     * @return the directory to store project reports
+     * @since 7.1
+     */
+    @OutputDirectory
+    @Incubating
```

Comment:
I think this directory may not be an input for these tasks since they already have separate output properties

Reply:
Fixed.